docs/docs/reference_home.md

```diff
@@ -1,3 +1,135 @@
 # Reference
+    
+`polars` provides a large number of functions for numerous data types and this
+can sometimes be a bit overwhelming. Overall, you should be able to do anything
+you want with `polars` by specifying the **data structure** you want to use and 
+then by applying **expressions** in a particular **context**.
+
+
+## Data structure
+
+As explained in some vignettes, one of `polars`' biggest strengths is the ability
+to choose between eager and lazy evaluation, that require respectively a 
+`DataFrame` and a `LazyFrame` (with their counterparts `GroupBy` and `LazyGroupby`
+for grouped data). 
+
+We can apply functions directly on a `DataFrame` or `LazyFrame`, such as `rename()`
+or `drop()`. Most (but not all!) functions that can be applied to `DataFrame`s
+can also be used on `LazyFrame`s.
+
+Another common data structure is the `Serie`, which can be considered as the 
+equivalent of R vectors in `polars`' world. Therefore, a `DataFrame` is a list of
+`Series`.
+
+Operations on `DataFrame` or `LazyFrame` are useful, but many more operations
+can be applied on columns themselves by using various **expressions** in different
+**contexts**.
+
+
+
+## Contexts
+
+A context simply is the type of data modification that is done. There are 3 types
+of contexts:
+
+* select and modify columns with `select()` and `with_columns()`;
+* filter rows with `filter()`;
+* group and aggregate rows with `groupby()` and `agg()`
+
+Inside each context, you can use various **expressions**. Some expressions cannot
+be used in some contexts. For example, in `with_columns()`, you can only apply
```

Comment:
of curiosity what are the general rules for other contexts? Is it the same for filter and groupby, but not for agg ?